src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -131,18 +158,25 @@ where
             let sstable_store = self.sstable_store.clone();
             let meta_clone = meta.clone();
             let policy = self.policy;
+            let uploading_size = self.uploading_size.clone();
+            let threshold = usize::MAX;
+            let uploading_sender = self.uploading_sender.clone();
             let upload_join_handle = tokio::spawn(async move {
-                if policy == CachePolicy::Fill {
-                    let sst = Sstable::new_with_data(table_id, meta_clone, data.clone())?;
-                    sstable_store.put(sst, data, CachePolicy::Fill).await
+                if let Some(sender) = uploading_sender && uploading_size.load(Ordering::Relaxed) > threshold {
+                    let (tx, rx) = oneshot::channel();
+                    sender.send(UploadRequest{
```

Comment:
If `uploading_size` exceed threshold, we shall wait it decreasing rather than send it to another thread?

Reply:
> If `uploading_size` exceed threshold, we shall wait it decreasing rather than send it to another thread?

another thread will do uploading tasks which it has received sequentially, so maximum uploading size is threshold + size of 1 uploading task